ICE/ICE.cpp

```diff
@@ -103,7 +167,15 @@ void ICE::Train(int sample_times, int negative_samples, double alpha, double alp
         {            
             v1 = vvnet.SourceSample();
             v2 = vvnet.TargetSample(v1);
-            vvnet.UpdatePair(w_vertex, w_context, v1, v2, dim, negative_samples, _alpha);
+
+            if (mode == "stage1"){
+                vvnet.UpdatePair(w_vertex, w_context, v1, v2, dim, negative_samples, _alpha);
+            }
+            else{
+                if (v1 >= vocab_count){
```

Comment:
what's the function of this condition? (you wanna sample a node with an offset vocab_count? if yes, I think we can use "v1+vocab_count" instead)